SlickEngagement_Plugin.php

```diff
@@ -516,37 +523,33 @@ private function echoPageBootData()
         }
 
         global $wp;
-        
-        $transient_name = $this->getTransientName(); //Name for WP Transient Cache API Key
+
+        $transient_name = $this->getTransientName(); // Name for WP Transient Cache API Key
+        $transient_name_ttl = $transient_name . '_ttl'; // Name for WP Transient Cache TTL Key
 
         // If `delete-boot=1` is passed as a query param, delete the stored page boot data
         $this->handleDeleteBootData();
 
         $no_transient_data = false === ($boot_data_obj = get_transient($transient_name)); //get_transient returns `false` if the key doesn't exist
+        $is_expired = get_transient( $transient_name_ttl ) < time();
 
         // If `slick-boot=1` is passed as a query param, force a re-fetch of the boot data from the server
         // If `slick-boot=0` is passed as a query param, skip fetching boot data from the server
         $slick_boot_param = $this->getQueryParamByName('slick-boot');
         $force_fetch_boot_data = ($slick_boot_param === '1');
         $dont_load_boot_data = ($slick_boot_param === '0');
         $slick_boot_param_not_set = ($slick_boot_param === null);
-    
-        // Check for existing data in transient cache. If none, then fetch data from server.
-        //TODO: store cache hits and cache misses in a transient or option?
-        if ($force_fetch_boot_data || ($no_transient_data && $slick_boot_param_not_set)) {
-            $this->echoSlickstreamComment("Fetching page boot data from server");
-            $boot_data_obj = $this->fetchBootData($siteCode);
-
-            // Put the results in transient storage; expire after 15 minutes
-            if ($boot_data_obj) {
-                set_transient($transient_name, $boot_data_obj, 15 * MINUTE_IN_SECONDS);
-                $this->echoSlickstreamComment("Storing page boot data in transient cache: " . $transient_name);
-            } else {
-                return;
-            }
-        } else if ($dont_load_boot_data) {
+
+        if ( $dont_load_boot_data ) {
             $this->echoSlickstreamComment("Skipping page boot data and CLS output");
             return;
+        }
+
+        if ( ( $force_fetch_boot_data || $no_transient_data || $is_expired ) ) {
+            if ( ! wp_next_scheduled( 'slickstream_fetch_boot_data', [ $siteCode, $transient_name, $transient_name_ttl ] ) ) {
+                $this->echoSlickstreamComment( 'Requesting page boot data via cron.' );
+                wp_schedule_single_event( time(), 'slickstream_fetch_boot_data', [ $siteCode, $transient_name, $transient_name_ttl ] );
```

Comment:
I'd consider having a single cron job that processes a queue of slipstream data, that would be more performant if there's going to be a lot of these jobs.